lib/systems/default.nix

```
@@ -83,6 +83,48 @@ let
       # TODO: deprecate args.rustc in favour of args.rust after 23.05 is EOL.
       rust = args.rust or args.rustc or { };
 
+      toolchain = {
+        cc =
+          if final.useLLVM || final.isDarwin then
+            "clang"
+          else if final.useArocc then
+            "arocc"
+          else if final.useZig then
+            "zig"
+          else
+            # TODO: don't imply GCC when nothing else is specified.
+            # Return null once we allow these attributes to be changed.
+            "gcc";
+
+        bintools =
+          if final.useLLVM || final.useArocc || final.useZig || final.isDarwin then "llvm" else "gnu";
+
+        cxxlib =
+          if final.useLLVM || final.useArocc || final.useZig || final.isDarwin then "libcxx" else "libstdcxx";
+
+        cxxrtlib =
+          if final.useLLVM || final.useArocc || final.useZig || final.isDarwin then
+            "libcxxabi"
+          else if final.isLinux then
+            "libsupc++"
+          else
+            "libcxxrt";
+
+        unwindlib =
+          if final.useLLVM || final.useArocc || final.useZig then
+            "libunwind"
+          else if final.isDarwin then
+            "libunwind-darwin-host"
```

```suggestion
            "libunwind-system"
```
“Host” is confusing wrt. `hostPlatform`. I think we don’t need to encode the Darwin detail because that can be conditioned on separately.